File: Java/src/main/java/org/sisuns/bucles_while/intermedio/Ejercicio_04.java
package org.sisuns.bucles_while.intermedio;

public class Ejercicio_04 {
    public static void main(String[] args) {
        // Declaración de variables y asignación de valores
        String texto = "Esta es una cadena de ejemplo para contar palabras";
        int contadorPalabras = 0;
        int longitud = texto.length();
        int i = 1;

        // Contar el número de palabras con un bucle while
        while (i <= longitud) {
            // Si encuentra un espacio, incrementar el contador de palabras
            if (texto.substring(i - 1, i).equals(" ")) {
                contadorPalabras++;
            }
            i++;
        }

        // Incrementar el contador para la última palabra (no seguida por espacio)
        contadorPalabras++;

        // Mostrar el resultado
        System.out.println("El número de palabras en el texto es: " + contadorPalabras);
    }
}

File: Python/Bucles_while/Intermedio/Ejercicio_04.py
# Lectura del texto desde la entrada estándar
texto = input("Ingrese una cadena de texto: ")

# Inicialización de variables
contador_palabras = 0
longitud = len(texto)
i = 0

# Contar el número de palabras con un bucle Mientras
while i < longitud:
    # Si encuentra un espacio, incrementar el contador de palabras
    if texto[i] == " ":
        contador_palabras += 1
    i += 1

# Incrementar el contador para la última palabra (no seguida por espacio)
contador_palabras += 1

# Mostrar el resultado
print("El número de palabras en el texto es:", contador_palabras)
